Error creating snatpool

Agent logs show following error when creating a snat pool:

2016-05-04 13:57:58.722 56098 ERROR f5_openstack_agent.lbaasv2.drivers.bigip.snats [req-33b4b334-9848-405b-bf7f-12b4f87f52e1 be3315095042428fb1ce7f5264f5bc61 d1aaf813eaa8455486b45f53ba467b71 - - -] Create SNAT pool failed 400 Unexpected Error: Bad Request for uri: https://10.50.50.21/mgmt/tm/ltm/snatpool/

Functional tests get numerous errors similar to these:
E       StateInvalidClient: Invalid state PENDING_DELETE of loadbalancer resource 75f6fdab-34ac-484f-8f93-6cf527263104
E       Neutron server returns request_ids: ['req-29f52c15-2ac2-4347-a40e-4c30585f8c70'] 
Or,
E       StateInvalidClient: Invalid state PENDING_UPDATE of loadbalancer resource bb35baee-a3bc-480a-955b-80ed1f4a04d4
E       Neutron server returns request_ids: ['req-86155c4e-965f-46fe-a4f4-c684bb65d322']

Reopening the issue.

See log tarball here:  https://console.aws.amazon.com/s3/home?region=us-west-2#&bucket=f5-openstack-testing-logs&prefix=LBaaSv2/lbaasv2-agent-issue-98/neutron_logs.tar

--- a/f5_openstack_agent/lbaasv2/drivers/bigip/snats.py
+++ b/f5_openstack_agent/lbaasv2/drivers/bigip/snats.py
@@ -220,9 +220,13 @@ class BigipSnatManager(object):
 
     def _delete_bigip_snats(self, bigip, subnetinfo, tenant_id):
         # Assure snats deleted in standalone mode """
-        # Assure snats deleted in standalone mode """
         subnet = subnetinfo['subnet']
-        partition = self.driver.service_adapter.get_folder_name(tenant_id)
+        network = subnetinfo['network']
+        if self.l2_service.is_common_network(network):
+            partition = 'Common'
+        else:
+            partition = self.driver.service_adapter.get_folder_name(tenant_id)
+        snat_pool_name = self.driver.service_adapter.get_folder_name(tenant_id)
         deleted_names = set()
         in_use_subnets = set()
         # Delete SNATs on traffic-group-local-only
@@ -251,7 +255,7 @@ class BigipSnatManager(object):
             LOG.debug('Remove translation address from tenant SNAT pool')
             try:
                 snatpool = self.snatpool_manager.load(bigip,
-                                                      index_snat_name,
+                                                      snat_pool_name,
                                                       partition)
 
                 snatpool.members = [

